clearbit/clearbit.go

```diff
@@ -18,13 +19,43 @@ type Client struct {
 	Reveal       *RevealService
 }
 
+// Config represents all the parameters available to configure a Clearbit
+// client
+type Config struct {
+	apiKey     string
+	httpClient *http.Client
+}
+
+// SetHTTPClient sets the optional http.Client we can use to make requests
+func SetHTTPClient(httpClient *http.Client) func(*Config) {
```

Comment:
instead of `Set` how do you feel about `With` (grabbed from here: https://github.com/grpc/grpc-go/blob/master/clientconn.go#L103-L107)

```go
client := clearbit.NewClient(
  clearbit.WithHTTPClient(&http.Client{}),
)
```